doc/src/manual/environment-variables.md

```diff
@@ -16,8 +16,13 @@ including those which include `JULIA` in their names.
 
 !!! note
 
-    Some variables, such as [`JULIA_NUM_THREADS`](@ref JULIA_NUM_THREADS) and [`JULIA_PROJECT`](@ref JULIA_PROJECT), need to be set before Julia
-    starts, therefore adding these to `~/.julia/config/startup.jl` is too late in the startup process.
+    Some variables, such as [`JULIA_NUM_THREADS`](@ref JULIA_NUM_THREADS) and
+    [`JULIA_PROJECT`](@ref JULIA_PROJECT), need to be set before Julia
+    starts. Adding these to `~/.julia/config/startup.jl` is too late in the startup process.
+
+    Also `__init__()` functions of modules in the sysimage will not see environment variables
+    that are set in `startup.jl` as they are run beforehand.
```

Comment:
```suggestion
    Also `__init__()` functions of modules may not see environment variables
    that are set in `startup.jl` as they are may be run beforehand.
```


E.g. if your startup.jl file does a `using`.

Reply:
I think the user can figure out if they're setting an ENV entry after a `using` in startup.jl. This is more about the sysimage gotcha

Reply:
Eh, generally speaking depending on the order seems iffy, and the sysimg has little to do with that.

Reply:
Maybe the advice should be generalized to avoid setting ENV in startup.jl, motivated by things like sysimage inits running before startup.jl

Reply:
It can also be worth noting that changing an environment variable at all can introduce data races into otherwise benign code, so is best carefully avoided as much as possible.